File: SWAIbot Simulator/SWAIbot Simulator Project/Assets/SWAIbot Simulator/Scripts/Protocol/Packet.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class Packet {
	
	private int targetId = 0;
	private int packetId = 0;
	
	private List<int> intList = new List<int>();
	private List<bool> boolList = new List<bool>();
	private List<string> strList = new List<string>();
	private List<short> shortList = new List<short>();
	
	public Packet(int targetId, int packetId) {
		this.targetId = targetId;
		this.packetId = packetId;
	}
	
	public int getTargetId() {
		return this.targetId;
	}
	
	public void setTargetId(int targetId) {
		this.targetId = targetId;
	}
	
	public int getPacketId() {
		return this.packetId;
	}
	
	public void setPacketId(int packetId) {
		this.packetId = packetId;
	}
	
	public List<int> getIntegers() {
		return this.intList;
	}
	
	public void appendInteger(int i) {
		intList.Add(i);
	}
	
	public bool hasIntegers() {
		if(intList == null)
			return false;
		
		return (intList.Count > 0);
	}
	
	public void setIntegers(List<int> intList) {
		this.intList = intList;
	}
	
	public void clearIntegers() {
		this.intList.Clear();
	}
	
	public List<bool> getBooleans() {
		return this.boolList;
	}
	
	public void appendBoolean(bool b) {
		boolList.Add(b);
	}
	
	public bool hasBooleans() {
		if(boolList == null)
			return false;
		
		return (boolList.Count > 0);
	}
	
	public void setBooleans(List<bool> boolList) {
		this.boolList = boolList;
	}
	
	public void clearBooleans() {
		this.boolList.Clear();
	}
	
	public List<string> getStrings() {
		return this.strList;
	}
	
	public void appendString(string str) {
		strList.Add(str);
	}
	
	public bool hasStrings() {
		if(strList == null)
			return false;
		
		return (strList.Count > 0);
	}
	
	public void setStrings(List<string> strList) {
		this.strList = strList;
	}
	
	public void clearStrings() {
		this.strList.Clear();
	}
	
	public List<short> getShorts() {
		return this.shortList;
	}
	
	public void appendShort(short s) {
		shortList.Add(s);
	}
	
	public bool hasShorts() {
		if(shortList == null)
			return false;
		
		return (shortList.Count > 0);
	}
	
	public void setShorts(List<short> shortList) {
		this.shortList = shortList;
	}
	
	public void clearShorts() {
		this.shortList.Clear();
	}
}

File: Eclipse Workspace/SWAIbot/src/com/timvisee/swaibot/protocol/packet/Packet.java
package com.timvisee.swaibot.protocol.packet;

import java.util.ArrayList;
import java.util.List;

public class Packet {

	private int targetId = 0;
	private int packetId = 0;

	private List<Integer> intList = new ArrayList<Integer>();
	private List<Boolean> boolList = new ArrayList<Boolean>();
	private List<String> strList = new ArrayList<String>();
	
	/**
	 * Constructor
	 * @param targetId Target device ID
	 * @param packetId Packet type ID
	 */
	public Packet(int targetId, int packetId) {
		this.targetId = targetId;
		this.packetId = packetId;
	}
	
	/**
	 * Get the target ID
	 * @return Target ID
	 */
	public int getTargetId() {
		return this.targetId;
	}
	
	/**
	 * Set the target ID
	 * @param targetId Target ID
	 */
	public void setTargetId(int targetId) {
		this.targetId = targetId;
	}
	
	/**
	 * Get the packet type ID
	 * @return Packet type ID
	 */
	public int getPacketId() {
		return this.packetId;
	}
	
	/**
	 * Set the packet type ID
	 * @param packetId Packet type ID
	 */
	public void setPacketId(int packetId) {
		this.packetId = packetId;
	}
	
	/**
	 * Get the list of integers in this packet
	 * @return List of integers
	 */
	public List<Integer> getIntegers() {
		return this.intList;
	}
	
	/**
	 * Append a new integer to the integer list in this packet
	 * @param i Integer to append
	 */
	public void appendInteger(Integer i) {
		intList.add(i);
	}
	
	/**
	 * Check whether this packet has any integers
	 * @return True if this packet had any integers
	 */
	public boolean hasIntegers() {
		if(intList == null)
			return false;
		
		return (intList.size() > 0);
	}
	
	/**
	 * Set the list of integers in this packet
	 * @param intList
	 */
	public void setIntegers(List<Integer> intList) {
		this.intList = intList;
	}
	
	/**
	 * Clear the list of integers
	 * @return Amount of integers cleared
	 */
	public int clearIntegers() {
		// Count the integer count
		int count =- this.intList.size();
		
		// Clear the integer list
		this.intList.clear();
		
		// Return element count
		return count;
	}
	
	/**
	 * Get the list of booleans in this packet
	 * @return List of booleans
	 */
	public List<Boolean> getBooleans() {
		return this.boolList;
	}
	
	/**
	 * Append a boolean to the booelan list
	 * @param b Boolean to append
	 */
	public void appendBoolean(Boolean b) {
		boolList.add(b);
	}
	
	/**
	 * Check whether this packet has any booleans
	 * @return True if this packet has any booleans
	 */
	public boolean hasBooleans() {
		if(boolList == null)
			return false;
		
		return (boolList.size() > 0);
	}
	
	/**
	 * Set the list of booleans
	 * @param boolList List of booleans
	 */
	public void setBooleans(List<Boolean> boolList) {
		this.boolList = boolList;
	}
	
	/**
	 * Clear the list of booleans
	 * @return Amount of booleans cleared
	 */
	public int clearBooleans() {
		// Get the boolean count
		int count = this.boolList.size();
		
		// Clear the booleans
		this.boolList.clear();
		
		// Return the amount of booleans cleared
		return count;
	}
	
	/**
	 * Get the list of strings in this packet
	 * @return List of strings
	 */
	public List<String> getStrings() {
		return this.strList;
	}
	
	/**
	 * Append a string to the string list
	 * @param str String to append
	 */
	public void appendString(String str) {
		strList.add(str);
	}
	
	/**
	 * Check whether this packet has any strings
	 * @return True if this packet has any strings
	 */
	public boolean hasStrings() {
		if(strList == null)
			return false;
		
		return (strList.size() > 0);
	}
	
	/**
	 * Set the list of strings in this packet
	 * @param strList String list
	 */
	public void setStrings(List<String> strList) {
		this.strList = strList;
	}
	
	/**
	 * Clear the list of strings
	 * @return Amount of strings cleared
	 */
	public int clearStrings() {
		// Count the strings
		int count = this.strList.size();
		
		// Clear the string list
		this.strList.clear();
		
		// Return the amount of strings cleared
		return count;
	}
}
